sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/TransactionCoordinator.java

```diff
@@ -0,0 +1,110 @@
+     * @param isCommit true for commit and false to rollback this transaction.
+     *
+     * @return a completable {@link Mono} which represent {@link DeliveryState}.
+     */
+    Mono<Void> completeTransaction(AmqpTransaction transaction, boolean isCommit) {
+        final Message message = Proton.message();
+        Discharge discharge = new Discharge();
+        discharge.setFail(!isCommit);
+        discharge.setTxnId(new Binary(transaction.getTransactionId().array()));
+        message.setBody(new AmqpValue(discharge));
+
+        final int payloadSize = messageSerializer.getSize(message);
+        final int allocationSize = payloadSize + MAX_AMQP_HEADER_SIZE_BYTES;
+
+        final byte[] bytes = new byte[allocationSize];
+        final int encodedSize = message.encode(bytes, 0, allocationSize);
+
+        return sendLink.send(bytes, encodedSize, DeliveryImpl.DEFAULT_MESSAGE_FORMAT, null)
+            .handle((outcome, sink) -> {
+                if (!(outcome instanceof Accepted)) {
+                    sink.error(new IllegalArgumentException("Expected a Accepted, received: " + outcome));
+                    return;
+                }
+            });
+    }
+
+    /**
+     * Creates the transaction in message broker.
+     *
+     * @return a completable {@link Mono} which represent {@link DeliveryState}.
+     */
+    Mono<AmqpTransaction> createTransaction() {
+        final Message message = Proton.message();
+        Declare declare = new Declare();
+        message.setBody(new AmqpValue(declare));
+
+        final int payloadSize = messageSerializer.getSize(message);
+        final int allocationSize = payloadSize + MAX_AMQP_HEADER_SIZE_BYTES;
+
+        final byte[] bytes = new byte[allocationSize];
+        final int encodedSize = message.encode(bytes, 0, allocationSize);
+
+        return sendLink.send(bytes, encodedSize, DeliveryImpl.DEFAULT_MESSAGE_FORMAT, null)
+            .handle((outcome, sink) -> {
+                if (!(outcome instanceof Declared)) {
+                    sink.error(new IllegalArgumentException("Expected a Declared, received: " + outcome));
+                    return;
+                }
+
+                final Declared state = (Declared) outcome;
+                final DeliveryState.DeliveryStateType stateType = state.getType();
+                switch (stateType) {
+                    case Accepted:
+                        break;
+                    case Rejected:
+                        break;
+                    case Declared:
+                        Binary txnId;
+                        Declared declared = (Declared) outcome;
+                        txnId = declared.getTxnId();
+                        logger.verbose("Created new TX started: {}", txnId);
+                        sink.next(new AmqpTransaction(txnId.asByteBuffer()));
+                        break;
+                    default:
+                        logger.warning("Unknown DeliveryState type: {}", stateType);
```

Comment:
Should the default behaviour just be a warning and not an error? the `send` operation will continue to wait for another delivery state message that will never come and hang an application.